src/pages/task/taskTabs/buildBaronAndAnnotations/AddIssueModal/index.tsx

```diff
@@ -67,64 +71,93 @@ export const AddIssueModal: React.VFC<Props> = ({
       title={title}
       onConfirm={handleSubmit}
-      submitDisabled={!state.canSubmit}
       buttonText={`Add ${issueString}`}
+      submitDisabled={!canSubmit}
     >
-      <StyledTextInput
-        data-cy="url-text-area"
-        label="URL"
-        value={state.url}
-        onChange={(e) => dispatch.setUrl(e.target.value)}
-        placeholder="https://example.com"
-        required
-        state={state.isURLValid ? "none" : "error"}
-        errorMessage="Please enter a valid URL"
-      />
-      <StyledTextInput
-        data-cy="issue-key-text-area"
-        label="Display Text"
-        value={state.issueKey}
-        onChange={(e) => dispatch.setKey(e.target.value)}
-        required
-        state={state.isKeyValid ? "none" : "error"}
-        errorMessage="Display Text must contain at least one character"
+      <SpruceForm
+        onSubmit={handleSubmit}
+        schema={addIssueModalSchema.schema}
+        uiSchema={addIssueModalSchema.uiSchema}
+        formData={formState}
+        onChange={({ formData, errors }) => {
+          setFormState(formData);
+          setCanSubmit(!errors.length);
+        }}
+        validate={handleURLValidation}
       />
-      <HR />
-      <Accordion title="Advanced Options">
-        <TextInput
-          data-cy="confidence-level"
-          label="Confidence Level"
-          description="The confidence level of the issue. This is a number between 0 and 100 representing a percentage."
-          value={state.confidenceScore}
-          optional
-          onChange={(e) => dispatch.setConfidenceScore(e.target.value)}
-          state={state.isConfidenceScoreValid ? "none" : "error"}
-          errorMessage="Please enter a number between 0 and 100"
-        />
-      </Accordion>
     </ConfirmationModal>
   );
 };
 
-const StyledTextInput = styled(TextInput)`
-  margin-bottom: ${size.s};
-`;
+const addIssueModalSchema: SpruceFormProps = {
+  schema: {
+    type: "object" as "object",
+    properties: {
+      url: {
+        type: "string" as "string",
+        title: "URL",
+      },
+      issueKey: {
+        type: "string" as "string",
+        title: "Display Text",
+      },
+      advancedOptions: {
+        type: "object" as "object",
+        properties: {
+          confidenceScore: {
+            type: ["number", "null"],
+            title: "Confidence Score",
+            minimum: 0,
+            maximum: 100,
+          },
+        },
+      },
+    },
+    required: ["url", "issueKey"],
+  },
+  uiSchema: {
+    url: {
+      "ui:data-cy": "issue-url",
+    },
+    issueKey: {
+      "ui:data-cy": "issue-key",
+    },
+    advancedOptions: {
+      "ui:ObjectFieldTemplate": AccordionFieldTemplate,
+      "ui:displayTitle": "Advanced Options",
+      confidenceScore: {
+        "ui:data-cy": "confidence-level",
+        "ui:description":
+          "The confidence score of the issue. This is a number between 0 and 100 representing a percentage.",
```

Comment:
You can use the optional flag to make it clear this field is not required
<img width="431" alt="image" src="https://user-images.githubusercontent.com/9298431/169845357-074a8a12-5c3b-470b-890f-b622806d3286.png">

```suggestion
          "The confidence score of the issue. This is a number between 0 and 100 representing a percentage.",
        "ui:optional": true,
```